atlp-rwanda/phantom-be-elite
# Ft assign bus to routes #13

What does this PR do?
Create, read and delete the assignment "assign a bus to route"

Description of Task to be completed?
As an operator, I want to assign a bus to a route so that they can control it.

The operator should be able to assign a bus to a route.
The operator should be able to view a paginated list of driver-to-bus assignments
The driver should receive a bus assignment and reassignment email & in-app notifications.

How should this be manually tested?
Clone the repo
checkout to branch` ft-assign-bus-to-routes-#13`
pull the changes by git pull
install all packages by running npm install
start the app by running `npm run start: dev`
In your API testing tool (postman is recommended for that) call API

Method: GET http://localhost:3000/api/v1/assign-route/ to view/read all assigned drivers.
Method: POST http://localhost:3000/api/v1/assign-route to create/register a new assign
Method: DELETE [http://localhost:3000/api/v1assign-route/{id}](http://localhost:5000/api/v1/assign/%7Bid%7D) to delete one specific assigned
Any background context you want to provide?
Note that you should send the request with the route, bus-number, and plate_number for creating a new assign in the request body

Format :
{ "route":"Mulindi - Remera", "bus_number":"Kimironko", "plate_number":"RAB 1230 V" }
What are the relevant pivotal tracker stories?
[Trello Board]https://trello.com/c/mHMeFgV1/13-assign-buses-to-routes

Screenshots (if appropriate)
N/A

Questions:
N/A
